src/System.Text.Json/src/System/Text/Json/Serialization/WriteStackFrame.cs

```diff
@@ -17,6 +18,10 @@ internal struct WriteStackFrame
         // Support Dictionary keys.
         public string KeyName;
 
+        // Whether the current object is an immutable dictionary.
+        public bool IsImmutableDictionary;
+        public bool IsPropertyAnImmutableDictionary;
```

Comment:
nit: should this be `IsImmutableDictionaryProperty` to align with similar naming on ReadStackFrame?